### PR TITLE
Fix issue 10082

### LINKED ISSUE #10082: Dynamic Routing: custom meta title & body_class output for static routes using data key
Reported in the forum: https://forum.ghost.org/t/when-using-dynamic-routes-meta-title-and-body-class-are-empty/3765

## Custom Meta Title

```
routes:
   /page1/:
    data: page.SLUG
    template: page
```

If you use

```
<head>
    <title>{{meta_title}}</title>
</head>

{{#page}}
{{/page}}
```

It always outputs the blog meta title and not the custom page meta title.

I see two problems:

1. the StaticRoutesRouter does not forward the correct context, [see](https://github.com/TryGhost/Ghost/blob/2.3.0/core/server/services/routing/StaticRoutesRouter.js#L85). It has to forward the resource type as well (e.g. `page`) if the `data` key was provided.

2. The meta [title helper](https://github.com/TryGhost/Ghost/blob/2.3.0/core/server/data/meta/title.js#L37) is not aware of the `this.page` context object -> `data.page`.

## Body Class Helper

```
{{body_class}}
```

This [helper](https://github.com/TryGhost/Ghost/blob/2.3.0/core/server/helpers/body_class.js#L15) also needs to respect `this.page`.

**Code suggestion:** 

> page = this.post && this.post.page ? {slug: this.post.slug} : this.page || false;
> classes.push('page-' + page.slug);

## Workaround

```
{{#page}}
  {{meta_title}}
{{/page}}
```

Ghost: master

### PR DESCRIPTION
**Pull request by cotko**
This should fix https://github.com/TryGhost/Ghost/issues/10082

Also some helpers like [`tags`](https://github.com/TryGhost/Ghost/blob/2.3.0/core/server/helpers/tags.js) and [`exceprt`](https://github.com/TryGhost/Ghost/blob/2.3.0/core/server/helpers/excerpt.js) are still not aware of the page context, I can fix them as well, if these changes till now are ok.

### CODE REVIEW

File: core/server/data/meta/amp_url.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
-var urlService = require('../../services/url'),
+const urlService = require('../../services/url'),
     getUrl = require('./url'),
     _ = require('lodash');
 
 function getAmplUrl(data) {
-    var context = data.context ? data.context : null;
+    let context = data.context ? data.context : null;
```
Comment (kirrg001) [inline core/server/data/meta/amp_url.js]: Could you please PR the ES6 changes in a separate PR?
The PR is hard to review now.

NOTE: We use constantly const in Ghost,

**Comment by cotko** [comment]
I'll add new tests, just have a question for this one: [keywords_spec](https://github.com/TryGhost/Ghost/blob/master/core/test/unit/data/meta/keywords_spec.js), it does not set the `context`, is there any case where `keywords` meta helper is used from within ghost without the context, or can context be added here as well, like it is [here](https://github.com/TryGhost/Ghost/blob/master/core/test/unit/data/meta/author_fb_url_spec.js#L8) and in others meta tests?

File: core/server/data/meta/amp_url.js (hunk 1 above)
Comment (cotko) [reply core/server/data/meta/amp_url.js]: Reverted for files that I did not change, is that ok, or do I do this for all?

Comment (kirrg001) [reply core/server/data/meta/amp_url.js]: Looks much better, thanks 👍

**Comment by kirrg001** [comment]
I try to look at your PR on Friday 👍

**Comment by kirrg001** [comment]
@cotko Will close this PR for now, because of no activity/reply.

---

In general: the underlying problem of #10082 is quite "big". Could you please create a PR per file change, otherwise it takes too long to review and test. Thanks so much!

**Pull request closed without merging**